### DISCUSSION TITLE
Booting in VirtualBox from CDRom using EFI

### DISCUSSION THREAD

**Discussion by MkStevo** [ravynsoft/ravynos#186] [General]
I found that I couldn't boot from the "virtual" ISO CDRom in VirtualBox. Initially I installed using a traditional BIOS but then found that I couldn't boot from the hard drive. Eventually I figured out how to boot the CDRom image.

Make sure the VirtualBox machine's Settings>System is set to EFI and the CDRom ISO image has been attached to the storage for the machine.

Boot the system, you'll probably find that you are left at the EFI shell. Type 'exit' (no quotes). You should be returned to the EFI "BIOS".
Select Boot Maintenance Management - press enter
Select Boot From File - press enter
Select the disk name that contains CDROM - press enter
Select [efi] - press enter (angle brackets are displayed around "efi")
Select [boot] - press enter (angle brackets are displayed around "boot")
Select Bootx64.efi - press enter

The CDRom should now boot and load airyxOS

If you install airyxOS to the hard drive you may find that you have to go through a similar procedure to boot from the hard drive image. The hard drive boot file appeared to be stored in a small partition with no volume label, it then was located in the efi\boot\bootx64.efi location. This can be selected as the first option to boot from using the EFI "BIOS" Boot Manager option and it will then be used every time the virtual machine is started. I didn't write the steps down for this, and it boots too quickly for me to try again!

[Made an edit replacing the angle brackets that surround "efi" and "boot" with square brackets as the angle brackets were interpreted as HTML and hid the words "efi" and "boot"]

**Comment by mszoek** [comment]
Aha! Good detective work here. BOOTX64.EFI is supposed to be the default loader. I guess not on VirtualBox? I'll make sure we set the loader name using `efitool` after installing to hopefully avoid any future issues. Can't do much about the ISO though since that is before any of our code runs.
